### PR TITLE
Update dns-01 record content to comply with ACME spec

### PR DESCRIPTION
**Pull request by jhass**
Following implementation reality in boulder, see https://github.com/letsencrypt/boulder/pull/1295

### CODE REVIEW

File: lib/acme/client/resources/challenges/dns01.rb [hunk 1]
```diff
@@ -12,6 +12,6 @@ def record_type
   end
 
   def record_content
-    crypto.digest.hexdigest(authorization_key)
+    Base64.urlsafe_encode64(crypto.digest.digest(authorization_key)).sub(/[\s=]*\z/, '')
```
Comment (unixcharles) [inline lib/acme/client/resources/challenges/dns01.rb]: what is the sub for exactly?

Comment (jhass) [inline lib/acme/client/resources/challenges/dns01.rb]: It drops the padding `urlsafe_encode64` adds, which Ruby's stdlib sadly has no option for.

Comment (unixcharles) [inline lib/acme/client/resources/challenges/dns01.rb]: ok

**Comment by dpiddy** [comment]
Would it be possible to get a gem cut with this fix included? Thanks!

**Comment by unixcharles** [comment]
@dpiddy bumped to 0.3.0

**Pull request merged**